Core/src/org/sleuthkit/autopsy/datasourcesummary/ui/ExcelExportAction.java

```diff
@@ -0,0 +1,310 @@
+
+    /**
+     * Main constructor.
+     *
+     * @param tabExports The different tabs that may have excel exports.
+     * @param dialogParent The component that jdialogs will claim as parent.
+     */
+    ExcelExportAction(List<? extends ExportableTab> tabExports, Component dialogParent) {
+        this.tabExports = Collections.unmodifiableList(new ArrayList<>(tabExports));
+        this.dialogParent = dialogParent;
+    }
+
+    /**
+     * Accepts the data source for which this export pertains, prompts user for
+     * output location, and exports the data.
+     *
+     * @param ds The data source.
+     */
+    @Override
+    public void accept(DataSource ds) {
+        if (ds == null) {
+            return;
+        }
+
+        File outputLoc = promptXLSXLocation(ds.getName());
+        if (outputLoc == null) {
+            return;
+        }
+        runXLSXExport(ds, outputLoc);
+    }
+
+    /**
+     * Prompts the user for an output location.
+     *
+     * @param dataSourceName The name of the data source.
+     * @return The file to which the excel document should be written or null if
+     * file already exists or cancellation.
+     */
+    @NbBundle.Messages({
+        "ExcelExportAction_promptAndExportToXLSX_fileExistsTitle=File Already Exists",
+        "# {0} - path",
+        "ExcelExportAction_promptAndExportToXLSX_fileExistsMessage=File at {0} already exists.",})
+    private File promptXLSXLocation(String dataSourceName) {
+        String expectedExtension = "xlsx";
+        JFileChooser fc = new JFileChooser();
+        FileNameExtensionFilter xmlFilter = new FileNameExtensionFilter("XLSX file (*.xlsx)", expectedExtension);
+        fc.addChoosableFileFilter(xmlFilter);
+        fc.setFileFilter(xmlFilter);
+
+        // set initial path to reports directory with filename that is 
+        // a combination of the data source name and time stamp
+        DateFormat dateFormat = new SimpleDateFormat("MM-dd-yyyy-HH-mm-ss");
+        String fileName = String.format("%s-%s.xlsx", dataSourceName == null ? "" : FileUtil.escapeFileName(dataSourceName), dateFormat.format(new Date()));
+        String reportsDir = "";
+        try {
+            reportsDir = Case.getCurrentCaseThrows().getReportDirectory();
+            File reportsDirFile = new File(reportsDir);
+            if (!reportsDirFile.exists()) {
+                reportsDirFile.mkdirs();
+            }
+        } catch (NoCurrentCaseException ex) {
+            logger.log(Level.WARNING, "Unable to find reports directory.", ex);
+        }
+
+        fc.setSelectedFile(Paths.get(reportsDir, fileName).toFile());
+
+        int returnVal = fc.showSaveDialog(dialogParent);
+
+        if (returnVal != JFileChooser.APPROVE_OPTION || fc.getSelectedFile() == null) {
+            return null;
+        }
+
+        File file = fc.getSelectedFile();
+
+        // insert xlsx if not at the end.
+        if (!file.getAbsolutePath().endsWith("." + expectedExtension)) {
```

Comment:
Strictly as FYI, FilenameUtils package has convenient methods to get file extension, name, etc. See example:
https://github.com/sleuthkit/autopsy/blob/develop/Experimental/src/org/sleuthkit/autopsy/experimental/autoingest/SupportedDataSources.java#L43